### PR TITLE
Issue/848 survey nongraded questions

### PR DESCRIPTION
**Pull request by zachberry**
* Updates babel config to prevent issue with tests not being transpiled correctly
* Adds survey type questions, which are questions that aren't graded

### CODE REVIEW

**Comment by zachberry** [comment]
Note: There is an issue still where the practice question in the visual editor renders incorrectly

File: packages/obonode/obojobo-chunks-question/viewer-component.test.js [hunk 1]
```diff
@@ -160,150 +201,124 @@ describe('Question', () => {
...
 
-		const component = renderer.create(<Question model={model} moduleData={moduleData} />)
-		const tree = component.toJSON()
+	test('Question in practice answered correctly', () => {
+		expect(createComponent('default', 'practice', 100, false).toJSON()).toMatchSnapshot()
+	})
 
-		expect(tree).toMatchSnapshot()
+	test('Survey question in practice not answered', () => {
+		expect(createComponent('survey', 'practice', null, false).toJSON()).toMatchSnapshot()
```
Comment (iturgeon) [inline packages/obonode/obojobo-chunks-question/viewer-component.test.js]: I'd really prefer to see a second expect that looks for the value expected specifically like `QuestionUtil.getScoreForModel.mockReturnValueOnce(100)`

File: packages/obonode/obojobo-sections-assessment/server/attempt-end.js [hunk 2]
```diff
@@ -183,12 +184,14 @@ const getCalculatedScores = (
...
 const calculateScores = (assessmentModel, attemptHistory, scoreInfo) => {
+	const numGradedQuestions = Object.keys(scoreInfo.scoresByQuestionId).length
+
 	const questionScores = scoreInfo.questions.map(question => ({
 		id: question.id,
 		score: scoreInfo.scoresByQuestionId[question.id] || 0
 	}))
 
-	const attemptScore = scoreInfo.scores.reduce((a, b) => a + b) / scoreInfo.questions.length
+	const attemptScore = questionScores.reduce((acc, s) => acc + s.score, 0) / numGradedQuestions
```
Comment (iturgeon) [inline packages/obonode/obojobo-sections-assessment/server/attempt-end.js]: Such a fitting use for `Array.reduce` here 🥇

**Comment by iturgeon** [comment]
Wanted to add a note, the addition of 'context' wasn't entirely clear what it meant.  I'm hypothesizing its the context a question is used in?  practice vs assessment?

General question too: essentially these survey questions are "questions used inside an assessment that collect student answers but have no bearing on the assessment scores"  never used outside of assessment? 

I bring these up mostly to brainstorm about making the concepts clear

**Comment by iturgeon** [comment]
fyi - this branch rebases cleanly

**Comment by iturgeon** [comment]
I'd like to suggest a squashing or amending to the commits on this branch, whole lotta nothing in the commit messages.

**Pull request merged**